hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/schema/MemberSchemaService.java

```diff
@@ -81,8 +81,8 @@ public Schema get(long schemaId) {
     }
 
     public CompletableFuture<Schema> getAsync(long schemaId) {
-        if (!nodeEngine.getClusterService().getClusterVersion().isEqualTo(Versions.V5_0)) {
-            throw new UnsupportedOperationException("The BETA compact format can only be used with 5.0 cluster");
+        if (!nodeEngine.getClusterService().getClusterVersion().isEqualTo(Versions.V5_1)) {
```

Comment:
Assuming a Hazelcast Enterprise cluster with mixed 5.0 & 5.1 members (so cluster operates at cluster version 5.0), this method will be throwing `UnsupportedOperationException`s from the 5.1 members. This doesn't seem like the desired behaviour, does it?

Reply:
@sancar can confirm but this was the behavior in our minds. We could do backward-incompatible changes to the wire level/implementation in 5.1, hence, only guarantee the same minor-version compatibility.

Reply:
ah, I see. Apparently I missed the all-capital `BETA` part ;-) thanks @mdumandag